aspnetcore/security/enforcing-ssl.md

```diff
@@ -10,45 +10,56 @@ ms.technology: aspnet
 ms.topic: article
 uid: security/enforcing-ssl
 ---
-# Enforce HTTPS in an ASP.NET Core
+# Enforce HTTPS in ASP.NET Core
 
 By [Rick Anderson](https://twitter.com/RickAndMSFT)
 
 This document shows how to:
 
-- Require HTTPS for all requests.
-- Redirect all HTTP requests to HTTPS.
+* Require HTTPS for all requests.
+* Redirect all HTTP requests to HTTPS.
 
 > [!WARNING]
-> Do **not** use `RequireHttpsAttribute` on Web APIs that receive sensitive information. `RequireHttpsAttribute` uses HTTP status codes to redirect browsers from HTTP to HTTPS. API clients may not understand or obey redirects from HTTP to HTTPS. Such clients may send information over HTTP. Web APIs should either:
+> Do **not** use [RequireHttpsAttribute](/dotnet/api/microsoft.aspnetcore.mvc.requirehttpsattribute) on Web APIs that receive sensitive information. `RequireHttpsAttribute` uses HTTP status codes to redirect browsers from HTTP to HTTPS. API clients may not understand or obey redirects from HTTP to HTTPS. Such clients may send information over HTTP. Web APIs should either:
 >
->* Not listen on HTTP.
->* Close the connection with status code 400 (Bad Request) and not serve the request.
+> * Not listen on HTTP.
+> * Close the connection with status code 400 (Bad Request) and not serve the request.
 
 <a name="require"></a>
 ## Require HTTPS
 
 ::: moniker range=">= aspnetcore-2.1"
-We recommend all ASP.NET Core web apps call `UseHttpsRedirection` to redirect all HTTP requests to HTTPS. If `UseHsts` is called in the app, it must be called before `UseHttpsRedirection`.
 
-The following code calls `UseHttpsRedirection` in the `Startup` class:
+We recommend all ASP.NET Core web apps call HTTPS Redirection Middleware ([UseHttpsRedirection](/dotnet/api/microsoft.aspnetcore.builder.httpspolicybuilderextensions.usehttpsredirection)) to redirect all HTTP requests to HTTPS. If HTTP Strict Transport Security (HSTS) Middleware ([UseHsts](/dotnet/api/microsoft.aspnetcore.builder.hstsbuilderextensions.usehsts)) is called in the app, it must be called before `UseHttpsRedirection`.
 
-[!code-csharp[sample](enforcing-ssl/sample/Startup.cs?name=snippet1&highlight=13)]
+The following code calls `UseHttpsRedirection` in the `Startup` class:
 
+[!code-csharp[](enforcing-ssl/sample/Startup.cs?name=snippet1&highlight=13)]
 
-The following code:
+The following code calls [AddHttpsRedirection](/dotnet/api/microsoft.aspnetcore.builder.httpsredirectionservicesextensions.addhttpsredirection) to configure middleware options:
 
-[!code-csharp[sample](enforcing-ssl/sample/Startup.cs?name=snippet2&highlight=14-99)]
+[!code-csharp[](enforcing-ssl/sample/Startup.cs?name=snippet2&highlight=14-99)]
 
-* Sets `RedirectStatusCode`.
+* Sets [HttpsRedirectionOptions.RedirectStatusCode](/dotnet/api/microsoft.aspnetcore.httpspolicy.httpsredirectionoptions.redirectstatuscode).
 * Sets the HTTPS port to 5001.
 
-::: moniker-end
+To specify an HTTPS port:
```

Comment:
Let's cover the automatic mechanisms before we talk about setting it manually.

- If you run Kestrel or HttpSys directly with HTTPS endpoints then the middleware can discover the ports via IServerAddressesFeature (so long as there's only one port).
- If you run behind a reverse proxy (including IIS/Express) then IServerAddressesFeature isn't available and the port must be configured.
- Visual Studio with set ASPNETCORE_HTTPS_PORT if IIS Express has HTTPS enabled.

You can configure it manually via:
- ASPNETCORE_HTTPS_PORT env var
- http_port config key (.json, command line, etc.)
- Set HttpsRedirectionOptions.HttpsPort in code.

Reply:
Perfect ... thanks. I'll get that in here.